tests/unit/test_index.py

```diff
@@ -30,8 +32,38 @@ def test_upsert_request_tuples_id_data_metadata(mocker):
                   ('vec2', vals2, md2)])
     index._vector_api.upsert.assert_called_once_with(
         pinecone.UpsertRequest(vectors=[
-            pinecone.Vector(id='vec1', values=vals1, metadata=md1),
-            pinecone.Vector(id='vec2', values=vals2, metadata=md2)
+            pinecone.Vector(id='vec1', values=vals1, metadata=md1, sparse_values={}),
+            pinecone.Vector(id='vec2', values=vals2, metadata=md2, sparse_values={})
+        ])
+    )
+
+
+def test_upsert_request_tuples_id_data_metadata_sparse_values(mocker):
+    import pinecone
+    pinecone.init(api_key='example-key')
+    index = pinecone.Index('example-name')
+    mocker.patch.object(index._vector_api, 'upsert', autospec=True)
+    index.upsert([('vec1', vals1, md1, sparse1),
+                  ('vec2', vals2, md2, sparse2)])
+    index._vector_api.upsert.assert_called_once_with(
+        pinecone.UpsertRequest(vectors=[
+            pinecone.Vector(id='vec1', values=vals1, metadata=md1, sparse_values=sparse1),
+            pinecone.Vector(id='vec2', values=vals2, metadata=md2, sparse_values=sparse2)
+        ])
+    )
+
+
```

Comment:
I would add some negative tests as well - for example a test where you pass a `Dict[int, float]` to the 3rd param in the tuple, and `assert` that a proper exception is raised